charger/daheimladen.go

```diff
@@ -0,0 +1,193 @@
+package charger
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/evcc-io/evcc/api"
+	"github.com/evcc-io/evcc/charger/daheimladen"
+	"github.com/evcc-io/evcc/util"
+	"github.com/evcc-io/evcc/util/request"
+	"golang.org/x/oauth2"
+)
+
+// DaheimLaden charger implementation
+type DaheimLaden struct {
+	*request.Helper
+	stationID     string
+	connectorID   int32
+	idTag         string
+	token         string
+	transactionID int32
+}
+
+func init() {
+	registry.Add("daheimladen", NewDaheimLadenFromConfig)
+}
+
+// NewDaheimLadenFromConfig creates a DaheimLaden charger from generic config
+func NewDaheimLadenFromConfig(other map[string]interface{}) (api.Charger, error) {
+	cc := struct {
+		Token     string
+		StationID string
+		IDTag     string
+	}{}
+
+	if err := util.DecodeOther(other, &cc); err != nil {
+		return nil, err
+	}
+
+	return NewDaheimLaden(cc.Token, cc.StationID, cc.IDTag)
+}
+
+// NewDaheimLaden creates DaheimLaden charger
+func NewDaheimLaden(token string, stationID string, idTag string) (*DaheimLaden, error) {
+	c := &DaheimLaden{
+		Helper:      request.NewHelper(util.NewLogger("daheim")),
+		stationID:   stationID,
+		connectorID: 1,
+		idTag:       idTag,
+		token:       token,
+	}
+	c.Timeout = 15
```

Comment:
Why is the timeout back now? Do you need more than 10 secs?

Reply:
Oh okay I guess I misunderstood, I thought your changes even with the Timeout in place so I added the timeout.
The internal RPCs have a default timeout of 10sec, so i kept the value at 15.
If you don't mind, when you can, can you try to run evcc with the timeout and without the timeout?
Or would you like me to delete line #52 altogether as its working for us?

Reply:
I've removed the line 52 for now as I realised you had gotten rid of it as well.